staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1166,6 +1166,137 @@ func TestCacherSendBookmarkEvents(t *testing.T) {
 	}
 }
 
+func TestInitialEventsEndBookmark(t *testing.T) {
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.WatchList, true)
+	forceRequestWatchProgressSupport(t)
+
+	backingStorage := &dummyStorage{}
+	cacher, _, err := newTestCacher(backingStorage)
+	if err != nil {
+		t.Fatalf("Couldn't create cacher: %v", err)
+	}
+	defer cacher.Stop()
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
+		if err := cacher.ready.wait(context.Background()); err != nil {
+			t.Fatalf("unexpected error waiting for the cache to be ready")
+		}
+	}
+
+	makePod := func(index uint64) *example.Pod {
+		return &example.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            fmt.Sprintf("pod-%d", index),
+				Namespace:       "ns",
+				ResourceVersion: fmt.Sprintf("%v", 100+index),
+			},
+		}
+	}
+
+	var currentResourceVersion uint64 = 3
+	createdPods := []*example.Pod{
+		makePod(1),
+		makePod(2),
+		makePod(currentResourceVersion),
+	}
+	for i := range createdPods {
+		if err := cacher.watchCache.Add(createdPods[i]); err != nil {
+			t.Fatalf("failed to add a pod: %v", err)
+		}
+	}
+
+	expectedWatchEventsFunc := func(createdPods []*example.Pod) []watch.Event {
+		var expectedWatchEvents []watch.Event
+		for _, pod := range createdPods {
+			expectedWatchEvents = append(expectedWatchEvents, watch.Event{Type: watch.Added, Object: pod})
+		}
+		return expectedWatchEvents
+	}
+
+	trueVal, falseVal := true, false
+
+	scenarios := []struct {
+		name                string
+		allowWatchBookmarks bool
+		sendInitialEvents   *bool
+		expectedWatchEvents []watch.Event
```

Comment:
expectedWatchEvents is the same in all cases  - let's remove it, see my comment above too